### PR TITLE
Gracefully shutdown on sigterm

### PR DESCRIPTION
**Pull request by frankh**

### CODE REVIEW

File: weaver/main.go [hunk 1]
```diff
@@ -98,5 +98,17 @@ func main() {
...
+	}
+
+	go log.Fatal(server.ListenAndServe())
+
+	sigChan := make(chan os.Signal)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+	<-sigChan
+	fmt.Println("Received sigterm, gracefully shutting down")
+	server.Shutdown(context.Background())
```
Comment (cedric-cordenier) [inline weaver/main.go]: +1 good code

Comment (MrSaints) [reply weaver/main.go]: oh really?

File: weaver/Dockerfile.build [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.7-alpine
+FROM golang:1.10-alpine
```
Comment (MrSaints) [inline weaver/Dockerfile.build]: :thinking:

Comment (frankh) [reply weaver/Dockerfile.build]: no 1.11 images have been pushed to docker hub

Comment (MrSaints) [reply weaver/Dockerfile.build]: :thinking:

File: weaver/main.go (hunk 1 above)
Comment (cedric-cordenier) [reply weaver/main.go]: yes, very graceful

Comment (frankh) [reply weaver/main.go]: turns out it isn't graceful at all and go sucks

**Pull request merged**